lib/oli_web/components/delivery/page_navigator.ex

```diff
@@ -0,0 +1,91 @@
+defmodule OliWeb.Components.Delivery.PageNavigator do
+  use Phoenix.LiveComponent
+
+  alias OliWeb.PageDeliveryView
+  alias OliWeb.Router.Helpers, as: Routes
+
+  attr :page_number, :integer, required: true
+  attr :next_page, :map
+  attr :previous_page, :map
+  attr :preview_mode, :boolean, default: false
+  attr :section_slug, :string, required: true
+  attr :numbered_revisions, :list, required: true
+  attr :show_navigation_arrows, :boolean, default: true
+  attr :id, :string, required: true
+
+  def render(assigns) do
+    assigns = assign(assigns, %{min_page: 1, max_page: assigns.numbered_revisions |> List.last() |> Map.get(:numbering_index, 1)})
+    ~H"""
+      <script>
+        function handleInputFocus_<%= @id %>(focus = true) {
```

Comment:
can we refactor this javascript to use Phoenix Hooks in a typescript module instead of rendering directly in the page?

Reply:
Never mind, Phoenix Hooks appear to only be supported within a LiveView